src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePoPCryptoProvider.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Security.Cryptography;
+using Microsoft.Identity.Client.AuthScheme.PoP;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.Platforms.netcore
+{
+    /// <summary>
+    /// This provider is based on the RSACryptoServiceProvider container concept. 
+    /// It will create and store a pair of keys in the context of the current user. Subsequent calls to this provider will reuse this key.
+    /// </summary>
+    /// <remarks>
+    /// Key creation and storage only works on Windows. See https://stackoverflow.com/questions/41986995/implement-rsa-in-net-core/42006084 for more details.
+    /// </remarks>
+    internal class NetCorePoPCryptoProvider : IPoPCryptoProvider, IDisposable
+    {
+        internal /* internal for test only */ const int RsaKeySize = 2048;
+        internal /* internal for test only */ const string ContainerName = "com.microsoft.msal";
+        private static RSA s_SigningKey;
+        private bool s_PersistKey;
+        private DateTime s_KeyTimeValidTo;
+        private readonly uint s_DefaultKeyExpirationTime = 60 * 60 * 8; // Eight Hours
+
+        private RSA RsaKey
+        {
+            get
+            {
+                if (s_PersistKey)
+                {
+                    if (CheckKeyExpiration())
+                    {
+                        s_SigningKey = RSA.Create();
+                        s_SigningKey.KeySize = RsaKeySize;
+                        s_KeyTimeValidTo = DateTime.Now + TimeSpan.FromSeconds(s_DefaultKeyExpirationTime);
```

Comment:
Please use only UTC times, i.e. DateTime.UtcNow

Reply:
Or use DateTimeOffset...

 In any case, DateTime represents regional time, but here you only care about instantenous time. Solution is to use DateTime.UtcNow or DateTimeOffset 

See https://stackoverflow.com/questions/4331189/datetime-vs-datetimeoffset for details